pkg/sql/opt/norm/testdata/rules/scalar

```diff
@@ -2441,3 +2441,15 @@ project
  └── projections
       ├── b.arr:5[1] [as=arr:8, outer=(5)]
       └── b.arr:5[2] [as=arr:9, outer=(5)]
+
+norm expect=NormalizeLikePattern
+SELECT i FROM a WHERE s LIKE '%%abc%%%%dddd'
```

Comment:
We should have additional tests for `NOT LIKE`, `ILIKE`, `NOT ILIKE`, `SIMILAR TO`, and `NOT SIMILAR TO`, and a test where the rule does not match.

nit: I'd also simplify this to `SELECT s FROM a WHERE s LIKE ...` or `SELECT s LIKE '...' FROM a`